mifosng-android/src/main/java/com/mifos/mifosxdroid/online/savingaccountsummary/SavingsAccountSummaryFragment.java

```diff
@@ -456,4 +484,107 @@ public void onDestroyView() {
         void doTransaction(SavingsAccountWithAssociations savingsAccountWithAssociations, String
                 transactionType, DepositType accountType);
     }
+
+    // Takes ScreenShot of Relative layout
+    public Bitmap layoutToImage() {
+        File dir = new File(Environment.getExternalStorageDirectory().getPath(), "Mifos Docs");
+        if (!dir.isDirectory()) {
+            dir.mkdirs();
+        }
+        RelativeLayout relativeLayout = rootView.findViewById(R.id.savings_account_summary_pdf_rl);
+        relativeLayout.setDrawingCacheEnabled(true);
+        relativeLayout.setDrawingCacheBackgroundColor(getResources().getColor(R.color.white));
+        relativeLayout.buildDrawingCache();
+        Bitmap bm = relativeLayout.getDrawingCache();
+        ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+        bm.compress(Bitmap.CompressFormat.JPEG, 100, bytes);
+        return bm;
+    }
+
+    //Converts image to pdf and deletes image
+    public void createPDF() {
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
+            Bitmap HeadBitmap = layoutToImage();
+            Bitmap ListBitmap = getWholeListViewItemsToBitmap();
+            PdfDocument pdfDocument = null;
+            pdfDocument = new PdfDocument();
+            PdfDocument.PageInfo pageInfo = new PdfDocument.PageInfo.Builder(
+                    HeadBitmap.getWidth() + 20, HeadBitmap.getHeight() +
+                    ListBitmap.getHeight() + 20, 1).create();
+            PdfDocument.Page page = pdfDocument.startPage(pageInfo);
+            page.getCanvas().drawBitmap(HeadBitmap, 0, 0, null);
+            page.getCanvas().drawBitmap(ListBitmap, 0, HeadBitmap.getHeight(), null);
+            ListBitmap.recycle();
+            pdfDocument.finishPage(page);
+            String filePath = directory + File.separator + tv_clientName.getText() + ".pdf";
+            File myFile = new File(filePath);
+            try {
+                pdfDocument.writeTo(new FileOutputStream(myFile));
+            } catch (Exception e) {
+                Toast.makeText(getContext(), "Error in saving PDF", Toast.LENGTH_SHORT).show();
+            }
+            pdfDocument.close();
+            Toast.makeText(getContext(), "Pdf saved to " + filePath,
+                    Toast.LENGTH_SHORT).show();
+        } else {
+            Toast.makeText(getContext(),
```

Comment:
Are you sure about this. Their isn't any way around?

Reply:
There are some limitations with sdk. Below this it won't work. This is the best way to inform user if feature is not available to user